virtcontainers/clh.go

```diff
@@ -877,461 +767,256 @@ func MaxClhVCPUs() uint32 {
-	logFile   *CLILogFile
-	vsock     *CLIVsock
-	cmdline   *CLICmdline
-	nets      *CLINets
-}
+	if err != nil {
+		return openApiClientError(err)
+	}
 
-func (d *DefaultCLIBuilder) AddKernelParameters(params []Param) {
+	c.Logger().Debugf("VM state after boot: %#v", info)
 
-	if d.cmdline == nil {
-		d.cmdline = &CLICmdline{}
+	if info.State != "Running" {
+		return fmt.Errorf("VM state is not created after create")
 	}
-	d.cmdline.params = append(d.cmdline.params, params...)
-}
-
-func (d *DefaultCLIBuilder) SetConsole(console *CLIConsole) {
-	d.console = console
-}
 
-func (d *DefaultCLIBuilder) SetCpus(cpus *CLICpus) {
-	d.cpus = cpus
+	return nil
 }
 
-func (d *DefaultCLIBuilder) SetDisk(disk *CLIDisk) {
-	d.disk = disk
-}
+func (clh *cloudHypervisor) addVSock(cid int64, path string) {
+	clh.Logger().WithFields(log.Fields{
+		"path": path,
+		"cid":  cid,
+	}).Info("Adding HybridVSock")
 
-func (d *DefaultCLIBuilder) SetFs(fs *CLIFs) {
-	d.fs = fs
+	clh.vmconfig.Vsock = append(clh.vmconfig.Vsock, chclient.VsockConfig{Cid: cid, Sock: path})
 }
 
-func (d *DefaultCLIBuilder) SetKernel(kernel *CLIKernel) {
-	d.kernel = kernel
-}
+func (clh *cloudHypervisor) addNet(e Endpoint) {
+	clh.Logger().WithField("endpoint-type", e).Debug("Adding Endpoint of type %v", e)
+	mac := e.HardwareAddr()
+	tapPath := e.NetworkPair().TapInterface.TAPIface.Name
+	clh.Logger().WithFields(log.Fields{
+		"mac": mac,
+		"tap": tapPath,
+	}).Info("Adding Net")
 
-func (d *DefaultCLIBuilder) SetMemory(memory *CLIMemory) {
-	d.memory = memory
+	// FIXME: This is required by CH
+	// remove after PR is merged:
+	// https://github.com/cloud-hypervisor/cloud-hypervisor/pull/480
+	ip := "0.0.0.0"
+	mask := "0.0.0.0"
+	clh.vmconfig.Net = append(clh.vmconfig.Net, chclient.NetConfig{Mac: mac, Tap: tapPath, Ip: ip, Mask: mask})
 }
 
-func (d *DefaultCLIBuilder) AddNet(net CLINet) {
-	if d.nets == nil {
-		d.nets = &CLINets{}
+// Add shared Volume using virtiofs
+func (clh *cloudHypervisor) addVolume(volume types.Volume) error {
+	if clh.config.SharedFS != config.VirtioFS {
+		return fmt.Errorf("shared fs method not supported %s", clh.config.SharedFS)
 	}
-	d.nets.networks = append(d.nets.networks, net)
-}
 
-func (d *DefaultCLIBuilder) SetRng(rng *CLIRng) {
-	d.rng = rng
-}
-
-func (d *DefaultCLIBuilder) SetSerial(serial *CLISerialConsole) {
-	d.serial = serial
-}
-
-func (d *DefaultCLIBuilder) SetVsock(vsock *CLIVsock) {
-	d.vsock = vsock
-}
-
-func (d *DefaultCLIBuilder) SetAPISocket(apiSocket *CLIAPISocket) {
-	d.apiSocket = apiSocket
-}
+	vfsdSockPath, err := clh.virtioFsSocketPath(clh.id)
+	if err != nil {
+		return err
+	}
 
-func (d *DefaultCLIBuilder) SetLogFile(logFile *CLILogFile) {
-	d.logFile = logFile
+	clh.vmconfig.Fs = []chclient.FsConfig{
+		{
+			Tag:       volume.MountTag,
+			CacheSize: int64(clh.config.VirtioFSCacheSize << 20),
+			Sock:      vfsdSockPath,
+			//TODO this is not expose to user
+			//TODO change if dax is set in config
+			NumQueues: 1,
```

Comment:
Need these "todo's" documented a bit more, with corresponding GH issues.

Reply:
@jcvenegas 1 queue is fine for now since virtio-fs is still under development regarding the multiqueue support.